Simplenote/NoteContentHelper.swift

```diff
@@ -0,0 +1,82 @@
+import Foundation
+
+// MARK: - NoteContentHelper
+//
+struct NoteContentHelper {
+
+    private init() {
+
+    }
+
+    /// Returns structure of the content: range of title and body
+    ///
+    /// - Parameters:
+    ///     - content: note content
+    ///
+    static func structure(of content: String?) -> (title: NSRange, body: NSRange) {
+        let titleRange = trimmedTextRange(in: content, startingFrom: 0, endAtNewline: true)
+        let bodyRange: NSRange
+        if titleRange.isNotFound {
+            bodyRange = .notFound
+        } else {
+            bodyRange = trimmedTextRange(in: content, startingFrom: NSMaxRange(titleRange), endAtNewline: false)
+        }
+
+        return (
+            title: titleRange,
+            body: bodyRange
+        )
+    }
+
+    private static func trimmedTextRange(in content: String?, startingFrom startLocation: Int, endAtNewline: Bool) -> NSRange {
+        guard let content = content else {
+            return .notFound
+        }
+        let fullRange = content.nsString.fullRange
```

Comment:
Since `content.nsString` is invoked in few other spots, maybe assigning that to a constant is worthy?